### ISSUE TITLE
Block indent always inserts spaces for indentation, even when configured to use tabs.

### ISSUE THREAD

**Issue by fake-name** [jupyter/notebook#7027]
## Description

When in a jupyter notebook, with indentation set to tab, if you select a block of code and hit tab to indent it, it inserts spaces. 

## Reproduce

In jupyter notebook (codemirror? Jupyter has too many poorly delineated components):

1. Set the indentation to tabs:  
![image](https://github.com/jupyter/notebook/assets/1401239/f240252d-edf9-44f6-a048-23b08e947bbe)
2. Select a block of code.
3. Hit Tab
4. You get spaces: 
![image](https://github.com/jupyter/notebook/assets/1401239/c2df69b4-3dd9-4485-b599-df2e4567611b)

## Expected behavior

If you have the editor set to tab mode (the correct way to indent things), I'd expect the block indentation operation to insert tabs.

Additionally, Notebook 7 seems to no longer have visible tab characters, whereas they are shown in 6 and/or "nbclassic". I'd call that a regression.

## Context

- Operating System and version: Server: Ubuntu 22.04. Client: Windows 10
- Browser and version: Vivaldi 6.1.3035.302 (Stable channel) (64-bit) 

- Jupyter Notebook version: 7.0.3

I don't think the rest of the template is relevant, since this seems to be a logic error rather then a bug in the sense of the code doing something it's not intended to do.

**Comment by jtpio** [comment]
Thanks @fake-name for reporting :+1: 

Would you be able to check if this is an issue in JupyterLab as well? (since Notebook 7 is based on JupyterLab)

**Comment by krassowski** [comment]
It works well in text editor in JupyterLab. I think the problem here is that this menu should not be shown when Notebook rather than Text editor is active. The underlying problem is the same as issues:
- https://github.com/jupyterlab/jupyterlab/issues/8332
- https://github.com/jupyterlab/jupyterlab/issues/13382
- https://github.com/jupyterlab/jupyterlab/issues/14372
where users assume "Text Editor" also means "Code editor in notebook" (but it does not)

In this sense it is a bug in Notebook because in JupyterLab there is no such an entry in the view menu when using a notebook.

**Comment by krassowski** [comment]
Oh, sorry it is there in JupyterLab too. I think that a good solutions would be to add a new "Cell Editor Indentation" which would toggle indentation for all types of cell (this is what `Auto Close Brackets` does if I remember correctly), and only show it notebook view, while hiding the "Text Editor Indentation".

@fake-name for now you should be able to adjust the indentation in "Settings Editor", under "CodeMirror" (to adjust them globally) or under "Notebook" to adjust it on per-cell-type basis.

**Comment by FeelUsM** [comment]
so, [how to change indentation settings in jupyter notebook?](https://stackoverflow.com/questions/77905486/how-to-change-indentation-settings-in-jupyter-notebook)

I try to change Settings->Text Editr Indentation, but it affects only the editor but not the notebook. This does not depend on whether I open the notebook in a separate browser tab or in Jupiterlab.

Where is notebook settings?

**Comment by fake-name** [comment]
Hey look, it's a year later and I'm still having issues with codemirror and tabs.

It really, really, REALLY wants you to use spaces. If you have code with tab indentation, it tries to convert _just the bit you're editing_ to spaces, leading to broken code.

**Comment by pavelkomarov** [comment]
Yeah, this is infuriating, honestly. I'm team \t, and the fact I can't copy and paste in my code from elsewhere, edit it a little, and have it run smoothly is ridiculous. Where even is this "Settings Editor", under "CodeMirror"?

**Comment by krassowski** [comment]
@pavelkomarov  For some reason I cannot upload images to GitHub right now, but see the screenshots in https://gist.github.com/krassowski/0c116275d60235ae2aff72e37e95d64a

> Rather then fix the menu so that tabs actually work, apparently the solution was to just delete the menu

No one deleted any menu.

**Comment by pavelkomarov** [comment]
THAT WORKS! Holy crap! I'd lost all hope.
